### PR TITLE
ARM64-SVE: Implement IF_SVE_DW_2A, IF_SVE_DW_2B, IF_SVE_EB_1B

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Part of #94549. Note that the preferred disassembly for `FMOV <Zd>.<T>, #0.0` is `MOV <Zd>.<T>, 0`, per the [SVE docs](https://docsmirror.github.io/A64/2023-06/fmov_dup_z_i.html): `FMOV` is a pseudo-instruction of `DUP`, which is aliased by `MOV`.

cstool output:
```
00C03825  mov   z0.b, #0
01C07825  mov   z1.h, #0
02C0B825  mov   z2.s, #0
03C0F825  mov   z3.d, #0
10702025  pext  p0.b, pn8[0]
31716025  pext  p1.h, pn9[1]
5272A025  pext  p2.s, pn0xA[2]
7373E025  pext  p3.d, pn0xB[3]
98742025  pext  { p8.b, p9.b }, pn0xC[0]
B9756025  pext  { p9.h, p10.h }, pn0xD[1]
DA74A025  pext  { p10.s, p11.s }, pn0xE[0]
FB75E025  pext  { p11.d, p12.d }, pn0xF[1]
```

JitDisasm output:
```
mov     z0.b, #0
mov     z1.h, #0
mov     z2.s, #0
mov     z3.d, #0
pext    p0.b, pn8[0]
pext    p1.h, pn9[1]
pext    p2.s, pn10[2]
pext    p3.d, pn11[3]
pext    { p8.b, p9.b }, pn12[0]
pext    { p9.h, p10.h }, pn13[1]
pext    { p10.s, p11.s }, pn14[0]
pext    { p11.d, p12.d }, pn15[1]
```

cc @dotnet/arm64-contrib

### CODE REVIEW

**Review by a74nh — APPROVED** [review]
LGTM

**Comment by amanasifkhalid** [comment]
@kunalspathak this change does have some TP impact. I suspect this is from me adding the `insScalableOpts` parameter to `emitIns_R_R_I`. Just pointing this out if we're looking for reasons to split out the SVE instructions into separate `emitIns` methods.

File: src/coreclr/jit/emitarm64.cpp [hunk 1]
```diff
@@ -18300,6 +18377,14 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
             break;
         }
 
+        case IF_SVE_EB_1B: // ........xx...... ...........ddddd -- SVE broadcast integer immediate (unpredicated)
+            // ins is MOV for this encoding, as it is the preferred disassembly, so pass FMOV to emitInsCodeSve
```
[the remark this answers is not in the document]
Comment (TIHan) [reply src/coreclr/jit/emitarm64.cpp:18381]: This is fine to me. Comment describes why so it's all good.

**Review by TIHan — APPROVED** [review]
LGTM. 

I wouldn't worry too much about the TP regressions as I think I solved a chunk of it in https://github.com/dotnet/runtime/pull/97739 by separating the SVE format cases in `emitOutputInstr` into their own function.

We could also separate out the SVE instruction cases in the `emitIns` functions into their own `emitInsSve` which could further help mitigate any TP regressions, but we don't have to do that now.

**Comment by amanasifkhalid** [comment]
Thanks for the reviews!

> I wouldn't worry too much about the TP regressions as I think I solved a chunk of it in https://github.com/dotnet/runtime/pull/97739 by separating the SVE format cases in emitOutputInstr into their own function.

I didn't realize we'd accumulated that much TP overhead from the SVE switch cases; >1% TP improvement is a lot...

> We could also separate out the SVE instruction cases in the emitIns functions into their own emitInsSve which could further help mitigate any TP regressions, but we don't have to do that now.

`emitOutputInstr` is probably the worst offender, but I'd be curious to see how much more we can claw back by separating out the `emitIns` methods.

**Comment by TIHan** [comment]
> I didn't realize we'd accumulated that much TP overhead from the SVE switch cases; >1% TP improvement is a lot...

The 1% TP improvement may not be accurate because it's running the ARM64 path on windows/x64 because it still counts the instructions generated for x64 too.

The TP regressions to really look at is the linux/arm64 running on linux/arm64, which improved modestly.

> I'd be curious to see how much more we can claw back by separating out the emitIns methods.

I'd be curious as well. I already did it to `emitIns_R_R_R_R` in the PR, but it didn't change the TP, probably because `emitIns_R_R_R_R` is not common.

**Pull request merged**